src/main/java/org/apache/solr/benchmarks/task/UrlCommandTask.java

```diff
@@ -0,0 +1,152 @@
+package org.apache.solr.benchmarks.task;
+
+import io.prometheus.client.Histogram;
+import org.apache.solr.benchmarks.BenchmarkContext;
+import org.apache.solr.benchmarks.BenchmarksMain;
+import org.apache.solr.benchmarks.ControlledExecutor;
+import org.apache.solr.benchmarks.beans.TaskByClass;
+import org.apache.solr.benchmarks.prometheus.PrometheusExportManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A task similar to the simple TaskType.command but with more control (concurrent/retries/rate/duration etc)
+ */
+public class UrlCommandTask extends AbstractTask<UrlCommandTask.ExecutionResult> {
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private final List<String> urls;
+  private final int maxRetry; //max retry on non zero exit code, no retry on exception
+  private final int RETRY_DELAY = 10000; //10 secs delay
+  private final String taskName;
+  private final UrlResponseListener listener;
+  private final AtomicInteger runCounter = new AtomicInteger();
+
+  public static UrlCommandTask buildTask(TaskByClass taskSpec, List<String> urls) {
+    if (taskSpec.durationSecs == null && taskSpec.rpm == null) { //then we assume only run per urls once only
+      return new UrlCommandTask(taskSpec, urls, (long)urls.size());
+    } else {
+      return new UrlCommandTask(taskSpec, urls, null);
+    }
+  }
+
+  private UrlCommandTask(TaskByClass taskSpec, List<String> urls, Long maxExecution) {
+    super(taskSpec, maxExecution);
+    this.urls = urls;
+    this.taskName = taskSpec.name;
+    maxRetry = (int) taskSpec.params.getOrDefault("max-retry", 0);
+    listener = new UrlResponseListener();
```

Comment:
nit: you could initialize this when you declare it at the top of the class

Reply:
Do you mean having `private final UrlResponseListener listenerlistener = new UrlResponseListener();` instead? 😊 

It's more of my personal preference, if something is of "default value" (of no logical significance), i usually just put it as field declaration/initializer. But for things that are a bit different from the assumed default, I like to put it in ctor more, since ctor is a bit easier to debug/locate than field declarations (which can be scattered here and there depending on dev preferences 😓 )